cli/flags.rs

```diff
@@ -51,129 +40,177 @@ Environment variables:
 /// not cause an error. I also think this is ok because missing any of the
 /// privileged flags is not destructive. Userland flag parsing would catch these
 /// errors.
-fn set_recognized_flags(
-  opts: &Options,
-  flags: &mut DenoFlags,
-  args: Vec<String>,
-) -> Result<Vec<String>, getopts::Fail> {
-  let mut rest = Vec::<String>::new();
-  // getopts doesn't allow parsing unknown options so we check them
-  // one-by-one and handle unrecognized ones manually
-  // better solution welcome!
-  for arg in args {
-    let fake_args = vec![arg];
-    match opts.parse(&fake_args) {
-      Err(getopts::Fail::UnrecognizedOption(_)) => {
-        rest.extend(fake_args);
-      }
-      Err(e) => {
-        return Err(e);
-      }
-      Ok(matches) => {
-        if matches.opt_present("help") {
-          flags.help = true;
-        }
-        if matches.opt_present("log-debug") {
-          flags.log_debug = true;
-        }
-        if matches.opt_present("version") {
-          flags.version = true;
-        }
-        if matches.opt_present("reload") {
-          flags.reload = true;
-        }
-        if matches.opt_present("allow-read") {
-          flags.allow_read = true;
-        }
-        if matches.opt_present("allow-write") {
-          flags.allow_write = true;
-        }
-        if matches.opt_present("allow-net") {
-          flags.allow_net = true;
-        }
-        if matches.opt_present("allow-env") {
-          flags.allow_env = true;
-        }
-        if matches.opt_present("allow-run") {
-          flags.allow_run = true;
-        }
-        if matches.opt_present("allow-all") {
-          flags.allow_read = true;
-          flags.allow_env = true;
-          flags.allow_net = true;
-          flags.allow_run = true;
-          flags.allow_read = true;
-          flags.allow_write = true;
-        }
-        if matches.opt_present("no-prompt") {
-          flags.no_prompts = true;
-        }
-        if matches.opt_present("types") {
-          flags.types = true;
-        }
-        if matches.opt_present("prefetch") {
-          flags.prefetch = true;
-        }
-        if matches.opt_present("info") {
-          flags.info = true;
-        }
-        if matches.opt_present("fmt") {
-          flags.fmt = true;
-        }
-
-        if !matches.free.is_empty() {
-          rest.extend(matches.free);
-        }
-      }
-    }
+fn set_recognized_flags(matches: ArgMatches, flags: &mut DenoFlags) {
```

Comment:
This could be `impl From<ArgMatches> for DenoFlags`

Reply:
Sure, haven't done refactor yet, rather switching the engine leaving the rest as is. Thanks :+1: